mmdet/models/seg_heads/base_semantic_head.py

```diff
@@ -0,0 +1,75 @@
+from abc import ABCMeta, abstractmethod
+
+import torch.nn.functional as F
+from mmcv.runner import BaseModule, auto_fp16, force_fp32
+
+from ..builder import build_loss
+from ..utils import upsample_like
+
+
+class BaseSemanticHead(BaseModule, metaclass=ABCMeta):
+    """Base module of Semantic Head.
+
+    Args:
+        num_classes (int): the number of classes.
+        init_cfg (dict): the initialization config.
+        loss_semantic (dict): the loss of the semantic head.
+    """
+
+    def __init__(self,
+                 num_classes,
+                 init_cfg=None,
+                 loss_semantic=dict(
+                     type='CrossEntropyLoss', ignore_index=-1,
+                     loss_weight=1.0)):
+        super(BaseSemanticHead, self).__init__(init_cfg)
+        self.loss_semantic = build_loss(loss_semantic)
+        self.num_classes = num_classes
+
+    @force_fp32(apply_to=('logits', ))
+    def loss(self, logits, gt_semantic_seg):
+        if logits.shape[-2:] != gt_semantic_seg.shape[-2:]:
+            logits = upsample_like(logits, gt_semantic_seg)
+        logits = logits.permute((0, 2, 3, 1))
+        # hard code here, minus one
+        gt_semantic_seg = gt_semantic_seg - 1
```

Comment:
We should do it in the class mapping process, see how mmseg deal with it. I also suspect this will make the 255 area becomes 254.

Reply:
The default pad value is 0 in CocoDataset.

Reply:
Then we should add an argument of the head, named `label_shift`, whose default value is 1, to avoid hardcode here.